@re-/assert/test/metaTests/testHelpers.ts

```diff
@@ -0,0 +1,110 @@
+import { strict } from "node:assert"
+import { fromHere, readJson, shell } from "@re-/node"
+import {
+    ExpressionStatement,
+    Project,
+    SourceFile,
+    SyntaxKind,
+    VariableStatement
+} from "ts-morph"
+
+const EXPECTED_RESULTS_JSON_PATH = fromHere("expectedResults.json")
+const PATH_TO_ASSERTIONS_DIR = fromHere(".reassert")
+
+const getExpressionStatements = (
+    functionName: string,
+    varStatements: VariableStatement[]
+) =>
+    varStatements
+        .find((statement) => statement.getText().includes(functionName))
+        ?.getDescendantsOfKind(SyntaxKind.ExpressionStatement)
+
+const getTestComment = (statement: ExpressionStatement, text: string) => {
+    const range = statement.getLeadingCommentRanges()
+    return text
+        .slice(range[0].getPos(), range[0].getEnd())
+        .replace("//", "")
+        .trim()
+}
+export const cleanUpSourceFile = async (sf: SourceFile, template: string) => {
+    sf.replaceWithText(template)
+    await sf.save()
+}
+export const getExpectedResult = (
+    statement: ExpressionStatement,
+    expectedData: Record<string, string>,
+    text: string
+) => expectedData[getTestComment(statement, text)]
+
+export const checkIfTestsContainErrors = (
+    statements: ExpressionStatement[],
+    expectedValues: Record<string, string>,
+    fullText: string
+) => {
+    let hasErrors = false
+    for (const statement of statements) {
+        const expected = getExpectedResult(statement, expectedValues, fullText)
+        try {
+            strict.deepEqual(statement.getText(), expected)
+        } catch (e) {
+            // @ts-ignore
+            console.log(`actual: ${e.actual} \nexpected: ${e.expected}`)
+            hasErrors = true
+            break
+        }
+    }
+    return hasErrors
+}
+export const checkThatBenchSnapGetsPopulated = (
+    statements: ExpressionStatement[]
+) => {
+    const meanMedianRegex = /\.(median|mean)\(`\d+\.\d+(ns|us|s|ms)`\)/
+    const markRegex =
+        /\.mark\(\{(mean|median): ?`\d+\.+\d+(ns|us|ms|s)\\, ?median: ?`\d+\.\d+(ns|us|ms|s)`\}\)/
+    let errors = false
+    for (const statement of statements) {
+        if (
+            !(
+                meanMedianRegex.test(statement.getText()) ||
+                markRegex.test(statement.getText())
+            )
+        ) {
+            errors = true
+            break
+        }
+    }
+    return errors
+}
+
+export const getTestFileData = async (
+    filename: string,
+    templateFilename: string,
+    precache = true
+) => {
+    const filePath = fromHere(`${filename}.ts`)
```

Comment:
I think ideally we should not even need for these files to exist before the test. We can just generate them from this function at some arbitrary path, add them to the ts-morph project, check them, then delete them. This would simplify the tests as you wouldn't have to worry about providing multiple paths and would avoid adding a couple redundant files to git.